src/auth/src/credentials/service_account.rs

```diff
@@ -12,15 +12,77 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+//! [Service Account] Credentials type.
+//!
+//! A service account is an account for an application or compute workload
+//! instead of an individual end user. The recommended practice is to use
+//! Google Default Credentials, which relies on the configuration of the Google
+//! Cloud system hosting your application ([Google Compute Engine (GCE)][gce-link],
+//! [Google Kubernetes Engine (GKE)][gke-link], [Cloud Run]) to authenticate
+//! your workload or application.  But sometimes you may need to create and
+//! download a [service account key], for example, to use a service account
+//! when running your application on a system that is not part of Google Cloud.
+//!
+//! Service account credentials are used in this latter case.
+//!
+//! You can create multiple service account keys for a single service account.
+//! When you create a service account key, the key is returned as a string, in the
+//! format described by [aip/4112]. This string contains an ID for the service
+//! account, as well as the cryptographic materials (an RSA private key)
+//! required to authenticate the caller.
+//!
+//! Therefore, service account keys should be treated as any other secret
+//! with security implications. Think of them as unencrypted passwords. Do not
+//! store them where unauthorized persons or programs may read them.
+//!
+//! The credentials in this module use [Self-signed JWTs] to bypass the
+//! intermediate step of exchanging client assertions for OAuth tokens.
+//!
+//! The types in this module allow you to retrieve access tokens, and
+//! can be used with the Google Cloud client libraries for Rust.
+//!
+//! While the Google Cloud client libraries for Rust default to
+//! using the types defined in this module, you may want to use these types directly
+//! when the service account key is obtained from Cloud Secret Manager or a similar service.
+//!
+//! Example usage:
+//!
+//! ```
+//! # use google_cloud_auth::credentials::service_account::{Builder, ServiceAccountKey};
+//! # use google_cloud_auth::credentials::Credential;
+//! # use google_cloud_auth::errors::CredentialError;
+//! # tokio_test::block_on(async {
+//! let service_account_key = serde_json::from_value::<ServiceAccountKey>(serde_json::json!({
+//! "client_email": "test-client-email",
+//! "private_key_id": "test-private-key-id",
+//! "private_key": "", // <-- Provide valid PKCS#8 PEM key here
+//! "project_id": "test-project-id",
+//! "universe_domain": "test-universe-domain",
+//! })).unwrap();
```

Comment:
This seems like a rather indirect way to initialize the data structure.  A plain `ServiceAccountKey { ... }` seems clear enough.